### DISCUSSION TITLE
from rST to Ema

### DISCUSSION THREAD

**Discussion by jaygray0919** [srid/neuron#665] [General]
Srid, may I ask you a high-level question (acknowledging that the devil is in the detail)?

    How easy would it be to migrate an rST/Sphinx website to Ema?

Here is the rST/Sphinx site using ReadTheDocs: https://nutrients.readthedocs.io/en/latest/index.html
Here is the GitHub repo used to build/rebuild the RTD site: https://github.com/gigster99/sph_doc
Here is a Sphinx-generated build on another domain: https://docs.ontomatica.io/index.html

We would like to have one core platform for our Neuron blog and our Sphinx documentation.
We would use Ema if we can manage the content migration and menu/navigation management from Neuron/Sphinx to Ema.

rST is a superset of markdown, but one that - as i remember - you've managed before in Neuron

One issue, which we are willing to work on, is the method to build menus to link pages.
The Sphinx `toc` (table of contents) file guides the generation of menus and links
In Neuron, we handled links on a note-by-note basis (one note at a time) - a bottom-up, incremental approach

We would like to have the best of both worlds - a systematic menu design/generator and an incremental design where, like Neuron, we can add nodes to a network with little effort.

/jay

**Comment by srid** [comment]
You can approach it in a couple of ways:

1. Convert your .rst to .md using Pandoc
2. Add reST support to Emanote

The latter is non-trivial effort and someone will have to volunteer it since I do no need reST support myself.

*Or*, you can ditch Sphinx in favour of other documentation generator (eg: [mdBook](https://rust-lang.github.io/mdBook/)) that uses Markdown. You would still convert your documentation to Markdown, but now you can use either of the tools, depending on the purpose.

**Reply by jaygray0919** [reply]
OK, we'll convert .rst to .md first (#1), and preview in mdBook
mdBook gives us the same table-of-contents structure (and menu generation) as supported in Sphinx
then we'll migrate the .md files to Emanote
we also have Neuron .md files and links that we'll migrate to Emanote
our goal: have a table-of-contents structure *and* the "hot link" properties featured in Emanote (incrementally adding content)
